.github/workflows/dotnet-publish.yml

```diff
@@ -2,42 +2,23 @@
 name: Publish NuGet package
 
 on:
-  workflow_dispatch:
   push:
-    branches:
-      - main
+    tags: 
```

Comment:
Do you have experience with this trigger?

I think maybe 

```
on:
  workflow_dispatch:  
  release:
    types: [published]
```

will be more suitable when we want to create the releases through github.